### PR TITLE
[QUARKS-123] adaptable polling source recipe

### PR DESCRIPTION
**Pull request by dlaboss**
Open for feedback.
Can't merge this PR until quarks supports the functionality here: https://github.com/apache/incubator-quarks/pull/86

### CODE REVIEW

File: site/recipes/recipe_adaptable_filter_range.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Adaptable filter behavior
+title: Changing a Filter's Range
 ---
```
Comment (queeniema) [inline site/recipes/recipe_adaptable_filter_range.md:3]: Please use sentence style capitalization as mentioned in [QUARKS-104](https://issues.apache.org/jira/browse/QUARKS-104): "Changing a filter's range"

File: site/recipes/recipe_adaptable_polling_source.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
+---
+title: Changing a Polled Source Stream's Period
```
Comment (queeniema) [inline site/recipes/recipe_adaptable_polling_source.md]: Please use sentence style capitalization: "Changing a polled source stream's period"

File: site/recipes/recipe_adaptable_polling_source.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+---
+
+The [Writing a Source Function](recipe_source_function.html) recipe introduced the basics of creating a source stream by polling a data source periodically.
```
Comment (queeniema) [inline site/recipes/recipe_adaptable_polling_source.md:5]: Please use sentence style capitalization: "Writing a source function"

File: site/recipes/recipe_adaptable_polling_source.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+Oftentimes, a user wants the poll frequency to be adaptable rather than static.  For example, an event such a sudden rise in a temperature sensor may motivate more frequent polling of the sensor and analysis of the data until the condition subsides.  A change in the poll frequency may be driven by locally performed analytics or via a command from an external source.
```
Comment (queeniema) [inline site/recipes/recipe_adaptable_polling_source.md]: Missing an "as" between "such" and "a" in: _For example, an event such a sudden rise..._

File: site/recipes/recipe_adaptable_polling_source.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+A Quarks IotProvider and IoTDevice with its command streams would be a natural way to control the application.  In this recipe we will just simulate a "set poll period" command stream.
```
Comment (queeniema) [inline site/recipes/recipe_adaptable_polling_source.md]: Consider putting `IotProvider` and `IotDevice` in inline code blocks

File: site/recipes/recipe_adaptable_polling_source.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+        // change the poll period using the control
+        System.out.println("Setting period="+period+" "+unit+" stream="+pollStream);
+        control.setPeriod(period, unit);
+    }
+```
+
+## Process the "set poll period" command stream
+
+Our commands are on the "TStream&lt;JsonObject&gt; cmds" stream.  Each JsonObject tuple is a command with the properties "period" and "unit".
```
Comment (queeniema) [inline site/recipes/recipe_adaptable_polling_source.md]: Consider putting `JsonObject` in an inline code block

File: site/recipes/recipe_adaptable_filter_range.md (hunk 1 above)
Comment (dlaboss) [inline site/recipes/recipe_adaptable_filter_range.md:3]: Will do.  Related: created QUARKS-153 (should have just added to 104 :-/)

File: site/recipes/recipe_adaptable_polling_source.md (hunk 2 above)
Comment (dlaboss) [inline site/recipes/recipe_adaptable_polling_source.md]: will do

File: site/recipes/recipe_adaptable_polling_source.md (hunk 3 above)
Comment (dlaboss) [inline site/recipes/recipe_adaptable_polling_source.md:5]: done

File: site/recipes/recipe_adaptable_polling_source.md (hunk 4 above)
Comment (dlaboss) [inline site/recipes/recipe_adaptable_polling_source.md]: done

File: site/recipes/recipe_adaptable_polling_source.md (hunk 5 above)
Comment (dlaboss) [inline site/recipes/recipe_adaptable_polling_source.md]: done

File: site/recipes/recipe_adaptable_polling_source.md (hunk 6 above)
Comment (dlaboss) [inline site/recipes/recipe_adaptable_polling_source.md]: done

**Pull request merged**